File: 1-100/1-10/1-twoSum/twoSum.cpp
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

class Solution {
  public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int, int> hashTable;
        for (size_t i = 0; i < nums.size(); ++i) {
            auto value = hashTable.find(target-nums[i]);
            if (value != hashTable.end()) {
                result.push_back(value->second);
                result.push_back(i);
            }
            
            hashTable.insert(make_pair(nums[i], i));
        }
        
        return result;
    }
};

File: 1-100/1-10/1-twoSum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        length = len(nums)
        hashmap = {}

        for i in range(length):
            if target - nums[i] in hashmap:
                return [hashmap[target-nums[i]], i]
            else:
                hashmap[nums[i]] = i

        return []